cmd/influxd/export/export.go

```diff
@@ -0,0 +1,367 @@
+	if err != nil {
+		return err
+	}
+
+	cmd.StdoutLogger.Printf("EXPORT: db=%s tsStart=%s tsEnd=%s",
+		cmd.database, tsStart, tsEnd)
+	if err := cmd.exportDatabase(tsStart, tsEnd); err != nil {
+		cmd.StderrLogger.Printf("export failed: %s", err)
+	}
+	return nil
+}
+
+// parseFlags parses and validates the command line arguments into a request object.
+func (cmd *Command) parseFlags(args []string) (tsStart, tsEnd time.Time, err error) {
+	fs := flag.NewFlagSet("", flag.ContinueOnError)
+
+	var startArg string
+	var endArg string
+
+	fs.StringVar(&cmd.host, "host", "localhost:8088", "")
+	fs.StringVar(&cmd.database, "database", "", "")
+
+	fs.StringVar(&startArg, "start", "", "")
+
+	fs.StringVar(&endArg, "end", "", "")
+
+	fs.SetOutput(cmd.Stderr)
+	fs.Usage = cmd.printUsage
+
+	err = fs.Parse(args)
+	if err != nil {
+		return
+	}
+	if startArg != "" {
+		tsStart, err = time.Parse(time.RFC3339, startArg)
+		if err != nil {
+			return
+		}
+	}
+
+	if endArg != "" {
+		tsEnd, err = time.Parse(time.RFC3339, endArg)
+		if err != nil {
+			return
+		}
+	}
+
+	// Ensure that only one arg is specified.
+	if fs.NArg() == 0 {
+		return time.Unix(0, 0), time.Unix(0, 0), errors.New("backup destination path required")
+	} else if fs.NArg() != 1 {
+		return time.Unix(0, 0), time.Unix(0, 0), errors.New("only one backup path allowed")
+	}
+	cmd.path = fs.Arg(0)
+
+	err = os.MkdirAll(cmd.path, 0700)
+
+	return
+}
+
+// exportShard will write a tar archive of the passed in shard with any TSM files that have been
+// created since the time passed in
+func (cmd *Command) exportShard(retentionPolicy string, shardID string, tsStart, tsEnd time.Time) error {
+	id, err := strconv.ParseUint(shardID, 10, 64)
+	if err != nil {
+		return err
+	}
+
+	shardArchivePath, err := cmd.nextPath(filepath.Join(cmd.path, fmt.Sprintf(BackupFilePattern, cmd.database, retentionPolicy, id)))
+	if err != nil {
+		return err
+	}
+
+	cmd.StdoutLogger.Printf("backing up db=%v rp=%v shard=%v to %s start %s end  %s",
+		cmd.database, retentionPolicy, shardID, shardArchivePath, tsStart, tsEnd)
+
+	req := &snapshotter.Request{
+		Type:            snapshotter.RequestShardExport,
+		Database:        cmd.database,
+		RetentionPolicy: retentionPolicy,
+		ShardID:         id,
+		ExportStart:     tsStart,
+		ExportEnd:       tsEnd,
+	}
+
+	// TODO: verify shard backup data
+	return cmd.downloadAndVerify(req, shardArchivePath, nil)
+}
+
+// exportDatabase will request the database information from the server and then backup the metastore and
+// every shard in every retention policy in the database. Each shard will be written to a separate tar.
+func (cmd *Command) exportDatabase(tsStart, tsEnd time.Time) error {
+	cmd.StdoutLogger.Printf("backing up db=%s start %s end %s", cmd.database, tsStart, tsEnd)
+
+	req := &snapshotter.Request{
+		Type:     snapshotter.RequestDatabaseInfo,
+		Database: cmd.database,
+	}
+	cmd.StdoutLogger.Println("made it to before the export response paths")
```

Comment:
This sounds like a debug print statement. Either remove it or update it to be a user-meaningful message